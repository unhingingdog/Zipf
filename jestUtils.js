import React from 'react'
import 'react-native'
import renderer from 'react-test-renderer'
import Adapter from 'enzyme-adapter-react-16'
import Enzyme, { shallow, render, mount } from 'enzyme'
Enzyme.configure({ adapter: new Adapter() })

global.React = React
global.renderer = renderer
global.shallow = shallow
