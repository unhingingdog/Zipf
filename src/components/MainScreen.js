import React, { Component } from 'react'
import {
	View,
	Text,
	TouchableOpacity,
	StyleSheet,
	Dimensions
} from 'react-native'
import { connect } from 'react-redux'
import { SubmitTextAction } from '../actions/SubmitTextAction'


export class MainScreen extends Component {
	constructor(props) {
		super(props)
		this.state = {
			orientation: 'portrait',
			clipboardMessage: ''
		}

		this.pasteButtonHandler = this.pasteButtonHandler.bind(this)
		this.onLayout = this.onLayout.bind(this)
		this.showClipboardMessage = this.showClipboardMessage.bind(this)
	}

	static navigationOptions = {
		header: null
	}

	onLayout(e) {
		const { width, height } = Dimensions.get('window')
		const orientation = width > height ? 'landscape' : 'portrait'
		this.setState({ orientation })
	}

	pasteButtonHandler() {
		const { dispatch } = this.props.navigation
		this.props.SubmitTextAction(dispatch)
	}

	showClipboardMessage() {
		const clipboardMessage = "(from your phone's clipboard)"
		setTimeout(() => { this.setState({ clipboardMessage }) }, 10000)
	}

	render() {
		if (!this.state.clipboardMessage) this.showClipboardMessage()
		return (
			<View style={styles.container} onLayout={this.onLayout}>
				<TouchableOpacity
					onPress={this.pasteButtonHandler}
					style={styles.pasteButton}
				>
					<Text style={styles.buttonText}>Paste</Text>
					<Text style={styles.subtext}>{this.state.clipboardMessage}</Text>
				</TouchableOpacity>
			</View>
		)
	}
}

const mapStateToProps = state => {
 return	{
		text: state.textInput.text
 	}
}

export default connect(mapStateToProps, {
	SubmitTextAction,
})(MainScreen)

const styles = StyleSheet.create({
	pasteButton: {
		backgroundColor: 'white',
		alignItems: 'center',
		margin: '10%'
	},
	container: {
		flex: 1,
		backgroundColor: 'white',
		justifyContent: 'center',
	},
	buttonText: {
		fontFamily: 'System',
		fontSize: 80,
		fontWeight: "100"
	},
	subtext: {
		color: 'grey'
	}
})
