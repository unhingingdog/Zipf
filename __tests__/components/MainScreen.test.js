import '../../jestUtils'
import { MainScreen } from '../../fakeComponents/MainScreen'
import TestRenderer from 'react-test-renderer'
import toJSON from 'enzyme-to-json'

describe('MainScreen renders correctly', () => {
  let text
  let TextChangedAction
  let SubmitTextAction
  let SubmitTextFromEditorAction
  let component
  let tree

  beforeEach(() => {
    TextChangedAction = jest.fn()
    SubmitTextAction = jest.fn()
    SubmitTextFromEditorAction = jest.fn()
    props = {
      text: 'test this shit',
      TextChangedAction,
      SubmitTextAction,
      SubmitTextFromEditorAction
    }
    component = shallow(<MainScreen {...props} />)
    tree = toJSON(component)
  })

  describe('default UI', () => {
    it('has the correct default state', () => {
      expect(component.state().inputType).toEqual('paste')
    })

    it('renders the default buttons', () => {
      expect(component.find("Button").length).toEqual(3)
      expect(component.find("PasteButton").length).toEqual(1)
    })
    it('matches the default snapshot', () => {
      expect(tree).toMatchSnapshot()
    })
  })
})
