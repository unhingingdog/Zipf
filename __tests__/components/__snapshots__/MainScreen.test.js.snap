// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainScreen renders correctly changes input mode to voice 1`] = `
<View>
  <PasteButton
    pasteWithButton={[Function]}
    textChange={[Function]}
    textValue="test this shit"
  />
  <Button
    className="textModeButton"
    onPress={[Function]}
    title="Input text"
  />
  <Button
    className="speakModeButton"
    onPress={[Function]}
    title="Speak"
  />
  <Button
    className="pasteModeButton"
    onPress={[Function]}
    title="paste from clipboard"
  />
</View>
`;

exports[`MainScreen renders correctly default UI matches the default snapshot 1`] = `
<View>
  <PasteButton
    pasteWithButton={[Function]}
    textChange={[Function]}
    textValue="test this shit"
  />
  <Button
    className="textModeButton"
    onPress={[Function]}
    title="Input text"
  />
  <Button
    className="speakModeButton"
    onPress={[Function]}
    title="Speak"
  />
  <Button
    className="pasteModeButton"
    onPress={[Function]}
    title="paste from clipboard"
  />
</View>
`;
